### DISCUSSION TITLE
Lego Remote - Mindstorm Robot Inventor Block Coding Question

### DISCUSSION THREAD

**Discussion by scbeme** [pybricks/support#1420] [Q&A]
I've just started learning Pybricks block coding to help my 8-year old grandson get started. Would appreciate suggestions on how to replace my many IF - IF ELSE statements with a more efficient solution. This current program functionally works but has a bit too much latency with my simple 2-wheeled car using the Mindstorm Robot Inventor hub. I'm hoping a simpler solution exists for reading the Lego remote control (7-switch version) switches into a list, and then using the 'list' values to command the two motors. Thanks in advance for any help you can provide. 

<img width="770" alt="Screenshot 2024-02-04 at 3 57 23 PM" src="https://github.com/pybricks/support/assets/118298647/5aa53253-67fd-4bcd-836e-d53d79729424">

basic_lego_remote_control_2_motors 2.py.zip

**Comment by laurensvalk** [comment]
Your program might work a lot faster if you remove the wait block that waits a whole second:

![image](https://github.com/pybricks/support/assets/12326241/2ad4dda0-9ddd-4ecc-aad6-99a6dba17a10)

-----------

Here is another way to do it, for further inspiration.

Instead of using many if-else blocks, this computes the speeds of the left and right motors individually, and then applies them as power levels directly to each motor.

![image](https://github.com/pybricks/support/assets/12326241/df388ec7-e802-4938-a180-26ea1d7d4a82)

**Reply by scbeme** [reply]
Thank you for the quick reply and recommendations! My apologies for not recognizing the 'obvious', ie, delete the 'wait' statement that I had initially added in case any troubleshooting was needed. duh...  After deleting, no more noticeable latency. My grandson will be thrilled to have a stand alone remote that doesn't require an intermediary computer. Plus I appreciate the additional and much simplified solution using differential wheel speeds vs versus my overly complicated code using the 'drive base' features. Eight year olds (and their grandparents) appreciate simplicity! 

I'm still curious if the 'get pressed' function can be implemented using block code?  I discovered the 'get pressed' code in the pybricks docs 'example robot' code. From my newbie understanding, 'get pressed' polls the pressed switches and stores them in a 'tuple' (which I'm just now beginning to investigate)? Seems this method would be very efficient for polling the switch positions and then being able to use the saved positions to direct the motors? 

Thank you again.

**Reply by laurensvalk** [reply]
Don't apologize, it simply means we don't have enough examples yet :smile: 

> My grandson will be thrilled to have a stand alone remote that doesn't require an intermediary computer.

Glad to hear it! Thank you for supporting this project by using the block coding feature, it's much appreciated.

> I'm still curious if the 'get pressed' function can be implemented using block code?

It could probably be implemented by calling Python code from blocks, but I wouldn't worry about efficiency for button checks. We do have an upcoming feature that lets you check if "any" button is pressed, which can be quite useful in some programs.

That way, you could choose to wait until at least one button is pressed, and then decide what to do based on which one.

**Reply by Beth57** [reply]
This post was helpful. Thank you. I too am trying to connect the remote to a simple robot for my grandson to use. I, however, am having trouble connecting the remote to the hub. I’m using a Spike Prime hub. The Lego remote 88010 does not want to connect to the hub. It connects fine to the Powered Up app. Should I post a separate thread for this problem in GitHub. I just signed up for GitHub and don’t really know what I’m doing.

**Comment by scbeme** [comment]
Are using Pybricks firmware with your Spike Prime hub, ie, you’ve installed the Pybricks firmware into your hub?

 Using our Robot Inventors hub with Pybricks, we’ve had no problems connecting the remote to the hub. 
 The steps we’ve been using…
- first we check the computer settings to ensure Bluetooth is turned on
 -start Pybricks
-load the program
-Turn on the Lego hub (rapid flashing blue led on the hub main / large button) and 9 LEDs  in center are solid white
-Click the Pybricks bluetooth icon (top left), a window pops up and displays your hub, select the hub in the pop up window , then click on PAIR (bottom right corner of pop up window)
-once your computer is Bluetooth paired with the hub, the bluetooth icon symbol  ‘shape’ is amended (slightly changed) and the hub blue led is now solid blue and the 9 LEDs are still solid white
-run the Pybricks program you coded by clicking the play icon (click on the triangle icon right of the Bluetooth icon)
-after clicking the play icon the hub large button is now slowly flashing blue and the 9 white LEDs are now ‘circling’ counter clockwise waiting on the remote control (the hub light pattern reverts to the previous after 8 seconds if the remote is not connected)
-click the remote control center green button within 2 seconds after clicking the play icon and in 1 or 2 (maybe 3) seconds the remote flashing white light will change to a steady blue light once connected

your remote should now work

Hope this helps.

**Reply by Beth57** [reply]
Yes that worked. Thank you very much. I didn’t know you had to connect the remote that quickly after downloading the program. 

Next question: is there any way of controlling the speed with this remote?

**Reply by msw-home** [reply]
Moin Berth57,
ich bin auch ein Anfänger mit meinem Enkel (68 + 7) Jahre und probiere so rum.
Die Geschwindigkeitsänderung habe ich wie folgt ausprobiert, geht bei uns. Sie Beispiel.
Viel Spaß noch beim testen und spielen.

![image](https://github.com/pybricks/support/assets/155910298/72f9814a-db63-47d7-8c45-f2d285af1021)

**Reply by Beth57** [reply]
Vielen danke. I learned some German as a child, but remember very little now. However I understood what you wrote. Wir werden das Programm ausprobieren
